policy/policy.go

```diff
@@ -251,6 +258,38 @@ func (rs *RuleSet) Or(rs1 *RuleSet) *RuleSet {
 	return &ruleSet
 }
 
+// mapRules returns a copy of the RuleSet where the given function has been applied to
+// all its child Rules.
+func (rs *RuleSet) mapRules(function func(*Rule) Rule) *RuleSet {
```

Comment:
I have no particular object to this, just commenting that my quick search did not reveal a collections framework.    But I might suggest that both mapRules and foreach might be useful publicly.  So rename mapRules Map and make foreach public?